internal/controller/nonadminbackup_controller.go

```diff
@@ -69,44 +72,98 @@ const (
 		return ctrl.Result{}, err
 	}
 
-	backupSpec, err := function.GetBackupSpecFromNonAdminBackup(&nab)
-
-	if backupSpec == nil {
-		logger.Error(err, "NonAdminBackup CR does not contain valid BackupSpec")
-		return ctrl.Result{}, nil
+	if nab.Status.Phase == constant.EmptyString {
+		// Phase: New
+		updatedStatus, errUpdate := function.UpdateNonAdminPhase(ctx, r.Client, logger, &nab, nacv1alpha1.NonAdminBackupPhaseNew)
+		if updatedStatus {
+			logger.V(1).Info("NonAdminBackup CR - Rqueue after Phase Update")
+			return ctrl.Result{Requeue: true, RequeueAfter: requeueTimeSeconds * time.Second}, nil
+		}
+		if errUpdate != nil {
+			logger.Error(errUpdate, "Unable to set NonAdminBackup Phase: New", nameField, req.Name, constant.NameSpaceString, req.Namespace)
+			return ctrl.Result{}, errUpdate
+		}
 	}
 
+	backupSpec, err := function.GetBackupSpecFromNonAdminBackup(&nab)
+
 	if err != nil {
-		logger.Error(err, "Error while performing NonAdminBackup reconcile")
+		errMsg := "NonAdminBackup CR does not contain valid BackupSpec"
+		logger.Error(err, errMsg)
+
+		// Phase: BackingOff
+		// BackupAccepted: False
+		// BackupQueued: False   # already set
+		updatedStatus, errUpdate := function.UpdateNonAdminPhase(ctx, r.Client, logger, &nab, nacv1alpha1.NonAdminBackupPhaseBackingOff)
+		if errUpdate != nil {
+			logger.Error(errUpdate, "Unable to set NonAdminBackup Phase: BackingOff", nameField, req.Name, constant.NameSpaceString, req.Namespace)
+			return ctrl.Result{}, errUpdate
+		} else if updatedStatus {
+			// We do not requeue as the State is BackingOff
+			return ctrl.Result{}, nil
+		}
+
+		updatedStatus, errUpdate = function.UpdateNonAdminBackupCondition(ctx, r.Client, logger, &nab, nacv1alpha1.NonAdminConditionAccepted, metav1.ConditionFalse, "InvalidBackupSpec", errMsg)
+		if updatedStatus {
+			return ctrl.Result{}, nil
+		}
+
+		if errUpdate != nil {
+			logger.Error(errUpdate, "Unable to set BackupAccepted Condition: False", nameField, req.Name, constant.NameSpaceString, req.Namespace)
+			return ctrl.Result{}, errUpdate
+		}
 		return ctrl.Result{}, err
 	}
 
-	logger.Info("NonAdminBackup Reconcile loop")
+	// Phase: New               # already set
+	// BackupAccepted: True
+	// BackupQueued: False   # already set
+	updatedStatus, errUpdate := function.UpdateNonAdminBackupCondition(ctx, r.Client, logger, &nab, nacv1alpha1.NonAdminConditionAccepted, metav1.ConditionTrue, "BackupAccepted", "backup accepted")
+	if updatedStatus {
+		return ctrl.Result{Requeue: true, RequeueAfter: requeueTimeSeconds * time.Second}, nil
+	}
+
+	if errUpdate != nil {
+		logger.Error(errUpdate, "Unable to set BackupAccepted Condition: True", nameField, req.Name, constant.NameSpaceString, req.Namespace)
+		return ctrl.Result{}, errUpdate
+	}
 
 	veleroBackupName := function.GenerateVeleroBackupName(nab.Namespace, nab.Name)
 
+	if veleroBackupName == constant.EmptyString {
```

Comment:
From line 154-line 214 seems like a task that we want the controller to perform only if the NAB.Status has `New` phase and `Backup Accepted` condition. Should'nt we be adding conditionals on these two things and then perform the task ? Also, IMO making a function of this whole Create Velero Backup task would be better suited.